projects/laji/src/app/+observation/result/observation-result.component.ts

```diff
@@ -150,46 +150,63 @@ export class ObservationResultComponent implements OnInit, OnChanges {
   }
 
   pickLocation(events: LajiMapDrawEvent[]) {
-    const query = {...this.query};
+    let value: Pick<WarehouseQueryInterface, 'coordinates' | 'polygonId'>;
+
     events.forEach(e => {
       let geometry: G.Geometry, layer: any;
       if (e.type === 'create') {
         geometry = e.feature.geometry;
         layer = e.layer;
-        // return;
       } else if (e.type === 'edit') {
         const keys = Object.keys(e.features);
         if (keys.length > 1) {
           throw new Error('Something wrong with map, there should never be multiple editable geometries');
         }
         geometry = e.features[keys[0]].geometry;
         layer = e.layers[keys[0]];
+      } else if (e.type === 'delete') {
+        if (e.features.length === 1) {
+          const newCoordinates = this.newQuery.coordinates || [];
+          const oldCoordinates = this.query.coordinates || [];
+          if (!Util.equalsArray(newCoordinates, oldCoordinates) || this.newQuery.polygonId !== this.query.polygonId) {
+            this.newQueryChange.emit({ ...this.newQuery, coordinates: this.query.coordinates, polygonId: this.query.polygonId });
+          }
+        } else if (e.features.length > 1) {
+          throw new Error('Something wrong with map, there should never be multiple deletable geometries');
+        }
+        return;
```

Comment:
Could help readibility, if the `e.features.length > 1` check would be on top, so there'd be no need for the other if block. The return statement would be more clear also since it would be inside the same block.